### PR TITLE
Add Alluxio Hive metastore

### PR DESCRIPTION
**Pull request by haoyuan**
```
Co-authored-by: David Zhu <david@alluxio.com>
Co-authored-by: Zac Blanco <zac@alluxio.com>
Co-authored-by: Calvin Jia <calvin@alluxio.com>
Co-authored-by: Bin Fan <binfan@alluxio.com>

== RELEASE NOTES ==

Hive Changes
This change introduces the Alluxio metastore which connects to the [Alluxio catalog service](https://docs.alluxio.io/os/user/2.1/en/core-services/Catalog.html).
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -735,12 +736,6 @@
                 <version>${dep.drift.version}</version>
             </dependency>
 
-            <dependency>
```
Comment (highker) [inline pom.xml]: This looks like an unrelated change

Comment (yuzhu) [reply pom.xml]: fixed

**Comment by yuzhu** [comment]
@highker I am working with @haoyuan on this PR. 
Fixed the compilation and pom files.  Could you please take a pass and give comments? 

Thanks. Also,  I guess I should go through the CLA process on EasyCLA?

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+import static com.facebook.presto.hive.metastore.MetastoreUtil.convertPredicateToParts;
+import static java.util.function.Function.identity;
+
+/**
+ * Implementation of the {@link HiveMetastore} interface through Alluxio.
+ */
+public class AlluxioHiveMetastore
+        implements ExtendedHiveMetastore
+{
+    private TableMasterClient client;
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: final

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+
+    @Inject
+    public AlluxioHiveMetastore(TableMasterClient client)
+    {
+        this.client = client;
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: requireNonNull

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+        }
+        catch (AlluxioStatusException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Set<ColumnStatisticType> getSupportedColumnStatistics(Type type)
+    {
+        throw new UnsupportedOperationException("getSupportedColumnStatistics");
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit: getSupportedColumnStatistics is not supported in AlluxioHiveMetastore.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+
+    @Override
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName)
+    {
+        try {
+            Table table = getTable(databaseName, tableName)
+                    .orElseThrow(() -> new PrestoException(HIVE_METASTORE_ERROR,
+                            String.format("Could not retrieve table %s.%s", databaseName, tableName)));
+            HiveBasicStatistics basicStats =
+                    ThriftMetastoreUtil.getHiveBasicStatistics(table.getParameters());
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit: put with the previous line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+    public Set<ColumnStatisticType> getSupportedColumnStatistics(Type type)
+    {
+        throw new UnsupportedOperationException("getSupportedColumnStatistics");
+    }
+
+    @Override
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName)
+    {
+        try {
+            Table table = getTable(databaseName, tableName)
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit
```java
Table table = getTable(databaseName, tableName).orElseThrow(() -> new PrestoException(
        HIVE_METASTORE_ERROR,
        String.format("Could not retrieve table %s.%s", databaseName, tableName)));

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+    }
+
+    @Override
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName,
+            String tableName, Set<String> partitionNames)
+    {
+        // TODO implement partition statistics
+        // currently returns a map of partitionName to empty statistics to satisfy presto requirements
+        return Collections.unmodifiableMap(
+                partitionNames.stream().collect(Collectors.toMap(identity(), (p) -> PartitionStatistics.empty())));
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit: put with the previous line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+    }
+
+    @Override
+    public void updateTableStatistics(String databaseName, String tableName,
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit: one param per line; keep the first line empty

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+            Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        throw new UnsupportedOperationException("updateTableStatistics");
+    }
+
+    @Override
+    public void updatePartitionStatistics(String databaseName, String tableName,
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: same

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+        // currently returns a map of partitionName to empty statistics to satisfy presto requirements
+        return Collections.unmodifiableMap(
+                partitionNames.stream().collect(Collectors.toMap(identity(), (p) -> PartitionStatistics.empty())));
+    }
+
+    @Override
+    public void updateTableStatistics(String databaseName, String tableName,
+            Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        throw new UnsupportedOperationException("updateTableStatistics");
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: same

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 11]
```diff
@@ -0,0 +1,397 @@
...
+    }
+
+    @Override
+    public void updatePartitionStatistics(String databaseName, String tableName,
+            String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
+    {
+        throw new UnsupportedOperationException("updatePartitionStatistics");
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: same

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+    }
+
+    @Override
+    public void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        throw new UnsupportedOperationException("dropColumn");
+    }
+
+    @Override
+    public Optional<Partition> getPartition(String databaseName, String tableName,
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: same

**Comment by haoyuan** [comment]
> @yuzhu, could you update the commit title/message to be
> 
> ```
> Add Alluxio Hive metastore
> ```

Done.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 13]
```diff
@@ -0,0 +1,384 @@
...
+ */
+public class AlluxioHiveMetastore
+        implements ExtendedHiveMetastore
+{
+    private final TableMasterClient client;
+
+    @Inject
+    public AlluxioHiveMetastore(TableMasterClient client)
+    {
+        requireNonNull(client, "client is null");
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit
```java
this.client = requireNonNull(client, "client is null");

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 14]
```diff
@@ -0,0 +1,384 @@
...
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
+    {
+        // TODO implement partition statistics
+        // currently returns a map of partitionName to empty statistics to satisfy presto requirements
+        return Collections.unmodifiableMap(partitionNames.stream().collect(Collectors.toMap(identity(), (p) -> PartitionStatistics.empty())));
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: We usually use `ImmutableMap.of`

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 15]
```diff
@@ -0,0 +1,384 @@
...
+            // Check that table name is correct
+            // TODO also check for database name equality
+            partitionInfos = partitionInfos.stream().filter(p -> p.getTableName().equals(tableName))
+                    .collect(Collectors.toList());
+            Map<String, Optional<Partition>> result = partitionInfos.stream()
+                    .filter(p -> partitionNames.stream().anyMatch(p.getPartitionName()::equals))
+                    .collect(Collectors.toMap(
+                            PartitionInfo::getPartitionName,
+                            pi -> Optional.of(ProtoUtils.fromProto(pi))));
+            return Collections.unmodifiableMap(result);
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: same

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 16]
```diff
@@ -0,0 +1,384 @@
...
+     * @param databaseName database name
+     * @param tableName    table name
+     * @param parts        list of values which returned partitions should contain
+     * @return optionally, a list of strings where each entry is in the form of {key}={value}
+     */
+    public Optional<List<String>> getPartitionNamesByParts(String databaseName, String tableName, List<String> parts)
+    {
+        try {
+            List<PartitionInfo> partitionInfos = ProtoUtils.toPartitionInfoList(
+                    client.readTable(databaseName, tableName, Constraint.getDefaultInstance()));
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: put this with the previous line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 17]
```diff
@@ -0,0 +1,384 @@
...
+            // TODO also check for database name equality
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: put an empty line before this commt

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 18]
```diff
@@ -0,0 +1,384 @@
...
+                        }
+                        for (int i = 0; i < values.size(); i++) {
+                            String constraintPart = parts.get(i);
+                            if (!constraintPart.isEmpty() && !values.get(i).equals(constraintPart)) {
+                                return false;
+                            }
+                        }
+                        return true;
+                    })
+                    .collect(Collectors.toList());
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: We usually use `toImmutableList`; same for all other places

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 19]
```diff
@@ -0,0 +1,384 @@
...
+        catch (AlluxioStatusException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Map<String, Optional<Partition>> getPartitionsByNames(String databaseName, String tableName, List<String> partitionNames)
+    {
+        if (partitionNames.isEmpty()) {
+            return Collections.emptyMap();
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: ImmutableMap.of

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 20]
```diff
@@ -0,0 +1,384 @@
...
+        }
+
+        try {
+            // Get all partitions
+            List<PartitionInfo> partitionInfos = ProtoUtils.toPartitionInfoList(
+                    client.readTable(databaseName, tableName, Constraint.getDefaultInstance()));
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: put this with the previous line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 21]
```diff
@@ -0,0 +1,384 @@
...
+            // Check that table name is correct
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: put an empty line before this commt

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 22]
```diff
@@ -0,0 +1,384 @@
...
+            // TODO also check for database name equality
+            partitionInfos = partitionInfos.stream().filter(p -> p.getTableName().equals(tableName))
+                    .collect(Collectors.toList());
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: put this with the previous line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 23]
```diff
@@ -0,0 +1,384 @@
...
+            Map<String, Optional<Partition>> result = partitionInfos.stream()
+                    .filter(p -> partitionNames.stream().anyMatch(p.getPartitionName()::equals))
+                    .collect(Collectors.toMap(
+                            PartitionInfo::getPartitionName,
+                            pi -> Optional.of(ProtoUtils.fromProto(pi))));
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit: `pi` -> `info`

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 24]
```diff
@@ -0,0 +1,384 @@
...
+            // Get all partitions
+            List<PartitionInfo> partitionInfos = ProtoUtils.toPartitionInfoList(
+                    client.readTable(databaseName, tableName, Constraint.getDefaultInstance()));
+            // Check that table name is correct
+            // TODO also check for database name equality
+            partitionInfos = partitionInfos.stream().filter(p -> p.getTableName().equals(tableName))
+                    .collect(Collectors.toList());
+            Map<String, Optional<Partition>> result = partitionInfos.stream()
+                    .filter(p -> partitionNames.stream().anyMatch(p.getPartitionName()::equals))
+                    .collect(Collectors.toMap(
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: immutablemap

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastoreConfig.java [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package com.facebook.presto.hive.metastore.alluxio;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+
+/**
+ * Configuration for the Alluxio compatible hive metastore interface.
+ */
+public class AlluxioHiveMetastoreConfig
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastoreConfig.java]: Add a test for this config

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java [hunk 26]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ProtoUtils
+{
+    private ProtoUtils() {}
+
+    public static com.facebook.presto.hive.metastore.Database fromProto(alluxio.grpc.table.Database db)
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java]: spell out db

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java [hunk 27]
```diff
@@ -0,0 +1,182 @@
...
+
+    public static Table fromProto(alluxio.grpc.table.TableInfo table)
+    {
+        if (!table.hasLayout()) {
+            throw new UnsupportedOperationException("Unsupported table metadata. missing layout.");
+        }
+        Layout layout = table.getLayout();
+        if (!alluxio.table.ProtoUtils.isHiveLayout(layout)) {
+            throw new UnsupportedOperationException("Unsupported table layout: " + layout);
+        }
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java]: put an empty line after this.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java [hunk 28]
```diff
@@ -0,0 +1,182 @@
...
+        try {
+            PartitionInfo partitionInfo = alluxio.table.ProtoUtils.toHiveLayout(layout);
+
+            // compute the data columns
+            Set<String> partitionColumns = table.getPartitionColsList().stream().map(FieldSchema::getName).collect(Collectors.toSet());
+            List<FieldSchema> dataColumns = table.getSchema().getColsList().stream().filter((f) -> !partitionColumns.contains(f.getName())).collect(Collectors.toList());
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java]: (f) -> field

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java [hunk 29]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    static StorageFormat fromProto(alluxio.grpc.table.layout.hive.StorageFormat format)
+    {
+        return StorageFormat.create(format.getSerde(), format.getInputFormat(), format.getOutputFormat());
+    }
+
+    public static StorageDescriptor fromProto(Storage storage)
+    {
+        StorageDescriptor sd = new StorageDescriptor();
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java]: spell out sd

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java [hunk 30]
```diff
@@ -0,0 +1,182 @@
...
+
+        return builder.build();
+    }
+
+    static StorageFormat fromProto(alluxio.grpc.table.layout.hive.StorageFormat format)
+    {
+        return StorageFormat.create(format.getSerde(), format.getInputFormat(), format.getOutputFormat());
+    }
+
+    public static StorageDescriptor fromProto(Storage storage)
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/ProtoUtils.java]: this method is not used?

**Comment by highker** [comment]
Once comment is address, feel free to click the "re-request review" button so I will get notified.

**Comment by yuzhu** [comment]
@highker Addressed your comment with the exception of adding test for `AlluxioHiveMetastore`

Regarding the test, I am planning to extend `AbstractTestHive` and use the docker containers to test the metastore. Does that sound like a reasonable plan to you?

**Comment by yuzhu** [comment]
I will also squash the commits once you have taken a pass. I want to separate the commits so it is easier to review.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java [hunk 31]
```diff
@@ -0,0 +1,73 @@
...
+        binder.bind(ExtendedHiveMetastore.class).to(AlluxioHiveMetastore.class).in(Scopes.SINGLETON);
+        newExporter(binder).export(ExtendedHiveMetastore.class).as(generatedNameOf(AlluxioHiveMetastore.class));
+    }
+
+    @Provides
+    @Inject
+    TableMasterClient provideCatalogMasterClient(AlluxioHiveMetastoreConfig config)
+    {
+        InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+        String addr = config.getMasterAddress();
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java]: `address`

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java [hunk 32]
```diff
@@ -0,0 +1,73 @@
...
+import static org.weakref.jmx.ObjectNames.generatedNameOf;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+/**
+ * Module for an Alluxio metastore implementation of the {@link HiveMetastore} interface.
+ */
+public class AlluxioMetastoreModule
+         extends AbstractConfigurationAwareModule
+{
+    private final String connectorId;
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java]: not used; so is the one in the constructor

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java [hunk 33]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+        String addr = config.getMasterAddress();
+        String[] parts = addr.split(":", 2);
+        conf.set(PropertyKey.MASTER_HOSTNAME, parts[0]);
+        if (parts.length > 1) {
+            conf.set(PropertyKey.MASTER_RPC_PORT, parts[1]);
+        }
+        MasterClientContext context = MasterClientContext
+                .newBuilder(ClientContext.create(new InstancedConfiguration(ConfigurationUtils.defaults()))).build();
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioMetastoreModule.java]: merge with the pervious line

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioProtoUtils.java [hunk 34]
```diff
@@ -0,0 +1,159 @@
...
+import com.facebook.presto.hive.metastore.SortingColumn;
+import com.facebook.presto.hive.metastore.StorageFormat;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.spi.security.PrincipalType;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioProtoUtils.java]: replace with Guava's stuff.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioProtoUtils.java [hunk 35]
```diff
@@ -0,0 +1,159 @@
...
+                .setParameters(info.getParametersMap())
+                .setValues(Lists.newArrayList(info.getValuesList()))
+                .setTableName(info.getTableName());
+
+        builder.getStorageBuilder()
+                .setSkewed(info.getStorage().getSkewed())
+                .setStorageFormat(fromProto(info.getStorage().getStorageFormat()))
+                .setLocation(info.getStorage().getLocation())
+                .setBucketProperty(info.getStorage().hasBucketProperty()
+                        ? fromProto(info.getStorage().getBucketProperty()) : Optional.empty())
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioProtoUtils.java]: merge into oneline

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java [hunk 36]
```diff
@@ -0,0 +1,380 @@
...
+            return ImmutableMap.of();
+        }
+
+        try {
+            // Get all partitions
+            List<PartitionInfo> partitionInfos = AlluxioProtoUtils.toPartitionInfoList(client.readTable(databaseName, tableName, Constraint.getDefaultInstance()));
+
+            // TODO also check for database name equality
+            partitionInfos = partitionInfos.stream().filter(p -> p.getTableName().equals(tableName)).collect(toImmutableList());
+            Map<String, Optional<Partition>> result = partitionInfos.stream()
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/alluxio/AlluxioHiveMetastore.java]: nit
```java
return partitionInfos.stream().
...

**Comment by yuzhu** [comment]
@highker thanks, i will fix those tonight. do you have any thoughts re:testing.

 I am planning to extend AbstractTestHive and use the docker containers to test the metastore. Does that sound like a reasonable plan to you?

**Comment by highker** [comment]
@yuzhu, if the test for the metastore module is too complicated, don't worry about that. The config test is good enough. Docker sounds an overkill.

**Comment by yuzhu** [comment]
@highker comments addressed, commits squashed. Let me know if there is anything else. 

Thanks!

**Comment by yuzhu** [comment]
@highker fixed. Thanks.
